Docker Installation

Hello,

I am now trying to get hublin up and running using Docker.  However I have had this error.  Does anyone have any ideas of what I am doing wrong?

Thanks

docker run -p 8080:8080 --link db:db --link redis:redis linagora/hublin
npm info it worked if it ends with ok
npm info using npm@3.10.9
npm info using node@v6.9.2
npm info lifecycle hublin@0.1.0~prestart: hublin@0.1.0
npm info lifecycle hublin@0.1.0~start: hublin@0.1.0

> hublin@0.1.0 start /src
> node server.js

Logger DailyRotateFile is disabled
Logger File is disabled
Warning: connect.session() MemoryStore is not
designed for a production environment, as it will leak
memory, and will not scale past a single process.
error: Fatal error: Error: Cannot find module 'easyrtc'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/src/modules/hublin-easyrtc-connector/backend/lib/EasyRTCAdapter.js:4:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at module.exports (/src/modules/hublin-easyrtc-connector/backend/lib/index.js:5:14)
    at lib (/src/modules/hublin-easyrtc-connector/index.js:17:38)
    at Promise.apply (/src/node_modules/q/q.js:1165:26)
    at Promise.promise.promiseDispatch (/src/node_modules/q/q.js:788:41)
error: Error: Cannot find module 'easyrtc'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/src/modules/hublin-easyrtc-connector/backend/lib/EasyRTCAdapter.js:4:15)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at module.exports (/src/modules/hublin-easyrtc-connector/backend/lib/index.js:5:14)
    at lib (/src/modules/hublin-easyrtc-connector/index.js:17:38)
    at Promise.apply (/src/node_modules/q/q.js:1165:26)
    at Promise.promise.promiseDispatch (/src/node_modules/q/q.js:788:41)

npm info lifecycle hublin@0.1.0~start: Failed to exec start script
npm ERR! Linux 4.4.0-53-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v6.9.2
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
npm ERR! hublin@0.1.0 start: `node server.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the hublin@0.1.0 start script 'node server.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the hublin package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node server.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs hublin
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls hublin
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /src/npm-debug.log

I have experienced the same issue (`Error: Cannot find module 'easyrtc'`), which is mainly due to a variance between the installation steps in the [README](https://github.com/linagora/hublin/blob/4db32d36c32a314f0e802f85d27430ffda660846/README.md) and the [Dockerfile](https://github.com/linagora/hublin/blob/4db32d36c32a314f0e802f85d27430ffda660846/Dockerfile). In this Dockerfile, there is a missing step:

> 7. Go into the modules directory and install easyrtc connector module dependecies
>
>        cd modules/hublin-easyrtc-connector
>        npm install 

This step actually installs the missing module `easyrtc` as it is a [dependency](https://ci.open-paas.org/stash/projects/MEET/repos/hublin-easyrtc-connector/browse/package.json?at=09a32086fe6190a77b39fc737592b3da4178b49c#44) of the submodule [hublin-easyrtc-connector](https://ci.open-paas.org/stash/scm/meet/hublin-easyrtc-connector.git).

For now and until the Dockerfile is fixed, a dirty workaround is to override the CMD with `/bin/sh -c "npm install ./modules/hublin-easyrtc-connector; npm start"`. For instance:

```bash
docker run -p 8080:8080 --link db:db --link redis:redis linagora/hublin /bin/sh -c "npm install ./modules/hublin-easyrtc-connector; npm start"
```
(should be adjusted according to your own workflow!)

--- a/Dockerfile
+++ b/Dockerfile
@@ -28,6 +28,8 @@ RUN cd /src && npm install --production --unsafe-perm
 
 ADD . /src
 
+RUN cd /src/modules/hublin-easyrtc-connector && npm install
+
 ADD config/db.json.docker /src/config/db.json
 
 ENV HUBLIN_REDIS_HOST redis

--- a/Dockerfile
+++ b/Dockerfile
@@ -28,7 +28,7 @@ RUN cd /src && npm install --production --unsafe-perm
 
 ADD . /src
 
-RUN cd /src/modules/hublin-easyrtc-connector && npm install
+RUN cd /src/modules/hublin-easyrtc-connector && npm install --production
 
 ADD config/db.json.docker /src/config/db.json
 

